### ISSUE TITLE
WebRemoteOrchestrator Error

### ISSUE THREAD

**Issue by ricardoamm** [Mimetis/Dotmim.Sync#808]
Hi, 

The same provisiong works for SqlSyncChangeTrackingProvider but not for WebRemoteOrchestrator.

![image](https://user-images.githubusercontent.com/26299584/189171748-828ca3bb-e9d3-4365-b373-5bfe3b23905a.png)

**Comment by Mimetis** [comment]
can you share your code ?

**Comment by ricardoamm** [comment]
`using System.Security.Claims;
using DevExpress.Xpo;
using Dotmim.Sync;
using Dotmim.Sync.Web.Server;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace com.smartthink.sigo.portal.Blazor.Server.Controllers;

[Authorize]
[ApiController]
[Route("api/[controller]")]
public class SyncController : ControllerBase
{
    private readonly IEnumerable<WebServerAgent> _webServerAgents;

    // Injected thanks to Dependency Injection
    public SyncController(IEnumerable<WebServerAgent> webServerAgents,
        IWebHostEnvironment env)
    {
        _webServerAgents = webServerAgents;
    }

    /// <Summary>
    ///     This POST handler is mandatory to handle all the sync process
    [HttpPost]
    public async Task Post()
    {
        // If you are using the [Authorize] attribute you don't need to check
        // the User.Identity.IsAuthenticated value
        if (HttpContext.User.Identity is { IsAuthenticated: true })
        {
            var uow = HttpContext.RequestServices.GetService<UnitOfWork>();

            var scopeName = HttpContext.GetScopeName();

            var webServerAgent = _webServerAgents.FirstOrDefault(
                c => c.ScopeName == scopeName);

            if (webServerAgent != null)
            {
                // OPTIONAL: -------------------------------------------
                // OPTIONAL: Playing with user coming from bearer token
                // OPTIONAL: -------------------------------------------

                // on each request coming from the client, just inject the User Id parameter
                webServerAgent.OnHttpGettingRequest(args =>
                {
                    if (args.Context.Parameters == null) return;
                    var pUserId = args.Context.Parameters["UserId"];

                    if (pUserId != null) return;
                    var userId =
                        HttpContext.User.Claims.FirstOrDefault(x => x.Type == ClaimTypes.NameIdentifier);
                    args.Context.Parameters.Add("UserId", userId);
                });

                // Because we don't want to send back this value, remove it from the response 
                webServerAgent.OnHttpSendingResponse(args =>
                {
                    if (args.Context.Parameters == null) return;
                    if (args.Context.Parameters.Contains("UserId"))
                        args.Context.Parameters.Remove("UserId");
                });

                await webServerAgent.HandleRequestAsync(HttpContext).ConfigureAwait(false);
            }
            else
            {
                HttpContext.Response.StatusCode = StatusCodes.Status406NotAcceptable;
            }
        }
        else
        {
            HttpContext.Response.StatusCode = StatusCodes.Status401Unauthorized;
        }
    }

    /// <summary>
    ///     This GET handler is optional. It allows you to see the configuration hosted on the server
    ///     The configuration is shown only if Environment == Development
    /// </summary>
    [HttpGet]
    [AllowAnonymous]
    public Task Get()
    {
        return HttpContext.WriteHelloAsync(_webServerAgents.FirstOrDefault());
    }
}`

`using System.Net;
using System.Net.Http.Headers;
using System.Text;
using com.smartthink.sigo.server.sync.setup;
using Dotmim.Sync;
using Dotmim.Sync.Enumerations;
using Dotmim.Sync.Sqlite;
using Dotmim.Sync.Web.Client;

namespace com.smartthink.sigo.portal.sync.client;

internal class Program
{
    private const string WebApiUrl = @"http://10.1.0.87/SIGO/api";
    private const string ClientConnectionString = @"C:\tmp\sigo.db";

    private const string Username = "cbSonils";
    private const string Password = "123";

    private static async Task Main()
    {
        var httpClient = new HttpClient();

        // Obtain a JWT token.
        var httpContent = new StringContent($@"{{ ""userName"": ""{Username}"", ""password"": ""{Password}"" }}",
            Encoding.UTF8, "application/json");
        var response =
            await httpClient.PostAsync($"{WebApiUrl}/Authentication/Authenticate", httpContent);

        if (response.StatusCode == HttpStatusCode.OK)
        {
            // Save the token for further requests.
            var token = await response.Content.ReadAsStringAsync();

            // Set the authentication header. 
            httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);

            // Adding the HttpClient instance to the web client orchestrator
            var serverOrchestrator = new WebRemoteOrchestrator($"{WebApiUrl}/sync", client: httpClient);

            // Second provider is using plain old Sql Server provider, relying on triggers and tracking tables to create the sync environment
            //var clientProvider = new SqlSyncProvider(SigoConnectionString);
            var clientProvider = new SqliteSyncProvider(ClientConnectionString);

            // Creating an agent that will handle all the process
            var syncOptions = new SyncOptions { UseVerboseErrors = true};

            // Creating an agent that will handle all the process
            var agent = new SyncAgent(clientProvider, serverOrchestrator, syncOptions);

            // Launch the sync process
            var parameters = new SyncParameters
            {
                { "\"filterCampBoss\"", $"{Username}" },
            };

            do
            {
                try
                {
                    var progress = new SynchronousProgress<ProgressArgs>(args =>
                        Console.WriteLine($"{args.ProgressPercentage:p}:\t{args.Message}"));

                    // Launch the sync process
                    var s1 = await agent.SynchronizeAsync("sigo_rh", parameters, progress);

                    // Write results
                    Console.WriteLine(s1);
                }
                catch (Exception e)
                {
                    Console.WriteLine(e.Message);
                }
            } while (Console.ReadKey().Key != ConsoleKey.Escape);
        }

        Console.WriteLine("End");
        Console.ReadKey();
    }
}`

**Comment by Mimetis** [comment]
I dont know how to reproduce your bug.
Your code seems legit, the bug is due to something else I can't see here
Need more details
Are you able to reproduce it with AdventureWorks ?

**Comment by ricardoamm** [comment]
Hello,

I use the SyncController with a single WebServerAgent and had no error, however the Bidirectional SyncDirection with the WebRemoteOrchestrator doesn't seems to work. The data is uploaded, but the changes in the server are not detected and the data is never downloaded, besides de first time. I made changes in the sql server tables, but they are never downloaded to the client.I tested with SqlSyncChangeTrackingProvider and the affected records are downloaded.
